README.md

```diff
@@ -1,2 +1,122 @@
+* Application infrastucture (keyboard shortcuts, commands, menus, layout), view
+  layer, and other utilities (signals, properties, messages) provided by phosphorjs
+  (http://phosphorjs.github.io/).
+* Models and views that can be used on any web page in the following repos:
+  - https://github.com/jupyter/jupyter-js-services
+  - https://github.com/jupyter/jupyter-js-terminal
+  - https://github.com/jupyter/jupyter-js-editor
+  - https://github.com/jupyter/jupyter-js-filebrowser
+  - https://github.com/jupyter/jupyter-js-input-area
+  - https://github.com/jupyter/jupyter-js-output-area
+  - https://github.com/jupyter/jupyter-js-cells
+  - https://github.com/jupyter/jupyter-js-notebook
+* Wrapping of those models and views as plugins for the Jupyter Workbench
+  - https://github.com/jupyter/jupyter-js-services-plugin
+  - https://github.com/jupyter/jupyter-js-metaservice-plugin
+  - https://github.com/jupyter/jupyter-js-output-area-plugin
+  - https://github.com/jupyter/jupyter-js-notebook-plugin
+  - https://github.com/jupyter/jupyter-js-terminal-plugin
+  - https://github.com/jupyter/jupyter-js-editor-plugin
+  - https://github.com/jupyter/jupyter-js-filebrowser-plugin
+* Movement towards having stable, documented JavaScript APIs.
+* Ability to hide/show code and entire cells based on query syntax.
+
+### Next+2
+
+* New plugins for the Jupyter Workbench:
+  - Standalone output area.
+  - Ability to hook kernels up to text editors and output areas.
+  - Themes.
+
+### Future
+
+* New plugins for the Jupyter Workbench:
+  - Notebook.
+  - Variable inspector.
+  - UI for managing plugins.
+* Dashboarding of output areas and interactive widgets.
+* Replace existing pages in the notebook web-application by the Jupyter Workbench.
+* Interactive debugger.
+* Real-time collaboration on notebooks, text-editors and other plugins.
+* Perform and publish accesibility audit by running an automated tool.
+* Bring our core plugins up to the level of the Web Accessibility Standards
+  (http://www.w3.org/standards/webdesign/accessibility)
+
+## ipywidgets
+
+### Next (4.1)
+
+* Improved and consistent visual style of built-in widgets.
+* Better abstractions for styling widgets using CSS and doing layout with flexbox.
+* Ability to run ipywidgets outside of the notebook.
+* Initial refactoring to start to use npm for packaging the JavaScript code.
+
+### Future
+
+* Further decoupling of the JavaScript and Python kernel backend.
+* Replace views based on backbone.js with phosphorjs widgets.
+* Wrap backbone models into standardized interfaces.
+* Transition to ES6/TypeScript.
+* Ability to pop-out widgets to the top-level of the Jupyter Workbench UI.
+
+## traitlets
+
+### Next (4.1)
+
+* Much nicer, decorator based APIs for observing (`@observe`), validating
+  (`@validate) and providing defaults (`@default) for traits.
+* Looking forward to its usage in Matplotlib.
+
+## nbconvert
+
+### Future
+
+* Implementations in either node.js or Pandoc.
```

Comment:
The place where a node.js implementation has come up is *only* for the html output. This would allow us to reuse the new npm packages we are creating and avoid having to have 2 implementations of the HTML DOM/CSS around. But for another other output format node.js makes 0 sense.